R/auto.R

```diff
@@ -93,27 +94,26 @@ pre_release_impl <- function(which, force) {
 }
 
 get_branch_name <- function() {
-  git2r::repository_head()$name
+  gert::git_branch()
 }
 
 get_remote_name <- function(branch) {
-  local <- git2r::branches(flags = "local")[[branch]]
-  upstream <- git2r::branch_get_upstream(local)
-  git2r::branch_remote_name(upstream)
+  branch_info <- gert::git_branch_list()
+  branch_info[branch_info$name == branch, ]$upstream
 }
 
 create_release_branch <- function(force) {
   branch_name <- paste0("cran-", desc::desc_get_version())
 
   cli_alert("Creating branch {.field {branch_name}}.")
 
-  git2r::branch_create(name = branch_name, force = force)
+  gert::git_branch_checkout(branch = branch_name, force = force)
```

Comment:
Do we need instead:

```suggestion
  if (gert::git_branch_exists(branch_name)) {
    if (force) {
      gert::git_branch_delete(branch_name)
    } else {
      abort(...)
    }
  }
  gert::git_branch_create(branch = branch_name, force = force)
```

?

Reply:
Seems legit, though this is possibly be not a minimal change ;-)

Reply:
If git2r supports overwriting branches, and gert doesn't, it's still a minimal change if we implement everything necessary to keep the original behavior. I added an assertion instead to unblock this.

Will release 1-2 more packages before merging this.